# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/arm/Assembler-arm.h
# Commit: b24505537d70
# Full Hash: b24505537d700cd4e01adee576645a617b6ef935
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 1: Reserve two slots after stack arguments for the future tls preservation. r=lth
#   
#   
#   We are going to remove Frame::tls and support trampolines for indirect calls, so we need to get rid of using Frame::tls.
#   In this and the followup patches I will iteratively remove all dependencies of Frame::tls and remove it eventually.
# ==============================================================================

diff -r 653e90061f03 -r b24505537d70 js/src/jit/arm/Assembler-arm.h
--- a/js/src/jit/arm/Assembler-arm.h	Fri Oct 30 08:31:42 2020 +0000
+++ b/js/src/jit/arm/Assembler-arm.h	Fri Oct 23 18:17:36 2020 +0000
@@ -147,6 +147,7 @@
   ABIArg next(MIRType argType);
   ABIArg& current() { return current_; }
   uint32_t stackBytesConsumedSoFar() const { return stackOffset_; }
+  void increaseStackOffset(uint32_t bytes) { stackOffset_ += bytes; }
 };
 
 bool IsUnaligned(const wasm::MemoryAccessDesc& access);